src/vos/vos_aggregate.c

```diff
@@ -511,15 +460,24 @@ prepare_segments(struct agg_merge_window *mw)
 			ent_in->ei_inob = 0;
 		}
 
+		if (cont_has_csums && ent_in->ei_inob != 0)
+			cs_total += vos_csum_prepare_ent(ent_in, phy_ent);
+
 		io->ic_seg_cnt++;
 		D_ASSERT(io->ic_seg_cnt <= io->ic_seg_max);
 	}
+	if (cont_has_csums && cs_total) {
+		rc = vos_csum_prepare_buf(io->ic_segs, io->ic_seg_cnt,
+				      &io->ic_csum_buf, io->ic_csum_buf_len,
+				      cs_total);
```

Comment:
Are you saying I should check the return value? This function returns 0, or -DER_NOMEM if it cannot allocate the buffer.  In this case it will be logged by the caller and the aggregation fail.
I can log the error here too.